lib/CL/clBuildProgram.c

```diff
@@ -307,6 +419,18 @@ CL_API_SUFFIX__VERSION_1_0
     pocl_cache_release_lock(program->read_locks[i]);
     program->read_locks[i] = NULL;
   }
+  if (program->num_kernels)
+    {
+      for (i=0; i < program->num_kernels; i++)
+        {
+          if (program->kernel_names)
+            POCL_MEM_FREE(program->kernel_names[i]);
+          if (program->default_kernels && program->default_kernels[i])
```

Comment:
Since clBuildProgram now pre-compiles every kernel in the program, i thought creating all kernels at this point could simplify some code that deals with kernels later, though i haven't worked on any of that for this PR. If you want i can remove it.